File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> answer(rowIndex+1);
        answer[0] = {1};
        for (int i=1; i<=rowIndex; i++) {
            answer[i].resize(i+1); // resize!!!!!!! 미친
            answer[i][0] = 1;
            answer[i][i] = 1;
            for (int j=1; j<i; j++) {
                answer[i][j] = answer[i-1][j-1] + answer[i-1][j];
            }
        }

        return answer[rowIndex];
    }
};

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        dp = [[1],[1,1]]
        if rowIndex == 0:
            return dp[0]
        if rowIndex == 1:
            return dp[1]
        idx = 2
        while idx <= rowIndex:
            l = [1] * (idx+1)
            for i in range(1, idx):
                l[i] = dp[idx-1][i-1] + dp[idx-1][i]
            dp.append(l)
            idx+=1

        return dp[rowIndex]